packages/apps/docs/src/scripts/importChangelogs/utils/reduce.ts

```diff
@@ -0,0 +1,75 @@
+export const reduceHeaderData = (header: IGitHubHeader): IChangelogHeader => {
+  const reduced: IChangelogHeader = {
+    'last-modified': header['last-modified'],
+    date: header.date,
+  };
+
+  return reduced;
+};
+export const reduceUserData = (user: IGitHubUser): IChangelogUser => {
+  const reduced: IChangelogUser = {
+    login: user.login,
+    id: user.id,
+    avatar_url: user.avatar_url,
+    url: user.url,
+  };
+
+  return reduced;
+};
+
+export const getAuthor = (
+  commit: IGithubCommitDataCommit,
+): IChangelogUser | undefined => {
```

Comment:
imo this `IChangelogUser` type could be improved because currently this makes it error prone because `id`, `url` and `avatar_url` not always available as part of the return data.

so making these fields as optional as part of return data it helps to safety check wherever we're consuming it

Reply:
I think I made an improvement

Reply:
you can remove the double casting from here https://github.com/kadena-community/kadena.js/pull/2063/files/be7a1835316c8e528a120d104ded95d6f060cb8b#diff-9475c0216adf8d4ed45a1637fbaf1c4aa3f19a1aa6df32ee145ae961f972242eR25-R29